AvicBot
===

(currently inactive)


(when me and abby went over the first 5 revisions, rev id's: 876773821, 877630206, 878854297, 878872180, 877568508)
* on AfCCSD page (what does that mean?), adds a user for empty page, later returns page to empty state

* on same page, adds draft one edit, removes 1 hour later

* on AvicBot/speedy page, adding and deleting various pages, categories, and drafts. unsure about the purpose of this list

* on AfCCSD page, empties page and empties again later

* adding and removing drafts from AfCCSD page.

note: looked at these pages and are no longer in use as of July 2020


do a write-up where we go deeper on some pairs - pick 2 self reverting and 1 bot-bot reverting combination -- each take one bot. can do in spreadsheet or notebook. go beyond bullet pointed list -- write a summary of why bot exists, what it does, why it gets reverted. give us a case study. what is the thing that is trying to be done here? deepen qualitative understanding

* why exist
* how long operational
* what percentage reverts
* why reverts itself
* namespaces
* link to code of bot? -- is publicly accessible?
* how does it work
* look at 10, 20 revisions
* look at 10, 20 reverted revisions

What does this bot do? How long has it been operational? Look at all the revisions this bot makes (10-20 revisions) and then 10-20 of its reverted revisions. What percentage of bots does it revert the most and get reverted the most? How does it interact with reverts? Go to talk pages 
Can’t find the code? (interesting to note) 
Require an understanding of the system

points not in order. collect as many links as you can. if find interesting rev, copy url and write why interesting. klepto approach - take as many as you can


Why AvicBot exist?
==

https://en.wikipedia.org/wiki/User:AvicBot

Source code is published, check the link above^^

AvicBots taks include: 
* maintaining interiwki links
* fixing redirects
* tagging uncategorized pages - adding a template that will add tagged pages to Category:Uncategorized pages or dated subcategories thereof.
* maintian a list of some categories - click the link above for exactly which caategories
* "category multiple merge targets" - you can merge multiple articles together and the target article is the one who oyou want to merge to - seems to be moving and nominating certian categories to this page, which are pages that "need attention and work by editors before the CfD request can be completed. - a similar process to the "articles for deletion" because a discussion is held determining whehter two or more articles shoud be merged
* maintaining {commonscat} category - used on roughly one percent of pages, this template {{Commons category}} links to a "Category:" name in name in the Wikimedia Commons repository that corresponds to the article or category.  If the Commons category name is misspelled or missing, the link will still appear "blue" even though the link will fail
* maintaining indef-blocked categories - user pages in this category are for users who have been indefinetely blocked and now oonly exist temporarily, usually to provide information to the users or allow them a suitable period of time to contest blocking. Indefinite does not mean "infinite" or "permanent". An indefinitely blocked user may later be unblocked in appropriate circumstances. In particularly serious cases in which no administrator would be willing to lift the block, the user is effectively banned by the community.
* tagging {{orphan}} pages - Template:Orphan is used on pages with no incoming links from other articles or lists (orphan). The template is responsible for populating sub-categories of Category:Orphaned articles ( a hidden category). There are currently 91,740 orphaned articles.
* merging 3+ {{multiple issues}} tags - This template tags articles or sections for multiple maintenance issues and then displays the alerts in a single box, as an alternative to using several space-consuming boxes, makes it mroe pleasant to look at, saves space.
* assisting with WP:CFD/W - CfD  is the central venue for discussing specific proposals to delete, merge, rename or split categories and stub types in accordance with the guidelines for categorization, category naming and stub articles.
* Maintaining CAT: UWCOI - "category: User talk pages with conflict of interest notcies" - this is a tracking category - It is used to build and maintain lists of pages. This category lists user talk pages that have received either a {{subst:Uw-coi}} or {{subst:Uw-coi-username}} notice. This information is used for spam and conflict of interests tracking. Talk pages are added automatically to this category via transclusion from either of the templates previously noted.

How long did AvicBot exist? 
==

the REquests for approval for AvicBot was filed on February 7 2011, however the result of the discussion was DENIED (https://en.wikipedia.org/wiki/Wikipedia:Bots/Requests_for_approval/AvicBot) and the fuinciton overview was to remove test edits and it eas denied because "This doesn't seem like a good task for a bot, or for an editor at bot-like speeds. The presence of such links in the article is a good sign that the history needs to be checked for more vandalism, not just "if the vandalism is right around the file". I do encourage you to go through these links and do the history checking to truly clean up the articles." There are multiple AvicBots though

Seems like most of the discussion for the request of bot approval went through a trial process for a certain number of edits, and once, the trial is complete, people leave their comments a member of the Bots Approval Group gives it the go ahead. Sometimes the results of the trial are posted to talk pages. Most of the edits that the trial is for is VERY small (all <60 edits)!!


* it seems that the number next to it doesn't represent a completely idfferent bot. It reperesents the discussion number for the requests of bot approval (or every time a bot wants/has another task 

 * 1. 2/7/2011 - DENIED - function overview: remove test edits - "This doesn't seem like a good task for a bot, or for an editor at bot-like speeds. The presence of such links in the article is a good sign that the history needs to be checked for more vandalism, not just 'if the vandalism is right around the file.'"
 * 2. 4/23/2011 - APPROVED - function overiew: find one category and replaces it with two, as needed; CATEGORY MULTIPLE MERGE TARGETS - 
 * 3. 7/3/2011 - APPROVED - function overview: standard interwiki bot, using pywikipedia; maintaing interwiki links- 
 * 4. 7/22/2011 (3:22) - WITHDRAWN BY OPERATOR - function overview: chaning links, update website links to new site, as there are no redirects, so all the old links are busted - withdrawn becuase "another user has taken up the task with AWB themselves." by Tamuz and closed by HELLKNOWZ
 * 5. 7/22/2011 (3:43PM) - APPROVED - function overview adding Cat interwikis/fixing double redirects; Maintain interwikis for category namespace, and fixing double redirects - "Looks good. Standard non-controversial tasks done by many other bots" by HELLKNOWZ (a member of Bot Approvals Group, an AfC reviewer ~ Articles of Creation.)
 * 6. 7/29/2011 - APPROVED - function overview: mainatin indef blocked categories; "Will grab users in the indef block categories, and check to see if they are, in fact, indef blocked. If a user is not found to be blocked, the username will be set aside on a subpage. If, after two weeks, the user is still in the category, and still unblocked, AvicBot will remove the category from the user talk page."
 * 7. 8/17/2011 - APPROVED - function overview: maintaining (and adding?) {{Commonscat}} templates; further details about the fucntion on this page: https://en.wikipedia.org/wiki/Wikipedia:Bots/Requests_for_approval/AvicBot_7
 * 8. 8/18/2011 - APPROVED - function overview: tagging uncategorized articles; "The bot will scan for Untagged Uncategorized Articles, and add {{Uncategorized|date={{subst:CURRENTMONTHNAME}} {{subst:CURRENTYEAR}}}} to them"
 * 9. 9/8/2011 - APPROVED - function overview: tag orphan pages: " Scans over pages in mainspace and finds orphans (Zero incoming links). If an Orphan page is found, it checks to see if the orphan tag (or "multiple issues|orphan=date" tags ) are on the page. It automatically skips redirects, {{wi}} pages, and disambig pages."
 * 10. 11/20/2011 - APPROVED - funciton overview: groupl multiple issues tags if there are 3 or more eligible tags; same logic as the AvicBot 9 task
 * 11. 5/27/2012 - APPROVED - function overview: performing category (re)moves as listed on WP:CFD/W; "AvicBot will either 1.) move pages from [[Caregory:foo]] to [[Category:bar]], as listed on Wikipedia:Categories for discussion/Working, if it's a move/merge request, or 2.) remove [[Caregory:foo]] entirely if the requests is to empty the category. Note that as AvicBot is not an AdminBot, it will not be deleting any pages." (SEEMS LIKE SOMETHING THAT WOULD CAUSE REVERTS OF ITS OWN EDITS). 
 * 12. 10/7/2015 - APPROVED - function overview: maintainign Category:User talk pages with conflict of interestr notices. Removing category after 6 months; "This will help to remove 'stale' reports and keep the category up to date and userful." (AN EXAMPLE OF A SELF-REVERT THAT MOST LIKELY NOT HAPPEN WITHIN 24 HORUS) 
 
 

How does it interact with reverts?
==

When lookin at the User contoributions for AvicBot, I put filtered the results to only show contributions from Jaunaury 1-21 in 2019, and when briefy scrollign thorugh, AvicBot has the SAME edit summaries: Bot: Listifying from Category:Candidates for speedy deletion, Candidates for speedy deletion as abandoned drafts or AfC submissions, Wikipedia usernames with possible policy issues, User talk pages with conflict of interest notices, Categories with incorrect underpopulated category template (which had the tag: blanking, which is most likely to cause a self-revert, if you're clearing the page), All articles proposed for deletion, AfC postponed G13, G13 eligible AfC submissions, Wikipedians who are indefinitely blocked for advertising

* Seems like AvicBot works a lot in tracking categories, but then again of course we don't know if these are all reverts

In [1]:
%reload_ext autoreload
%autoreload 2
%matplotlib inline

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# other imports can go here
from datetime import datetime
import pytz
import tqdm

In [4]:
df = pd.read_csv('revs_scored_jan.tsv', sep='\t', header=0)

In [5]:
days = df.rev_timestamp.map(lambda ts: datetime.utcfromtimestamp(ts).day)
df['day'] = days

In [6]:
# first thing: isolate data to only data from Jan1-Jan21, reverted w/in 24 hours, revisions by a bot

sdf = df[df.day<22]
sdf_reverted = sdf[sdf.is_reverted] # w/o, still include 'none'
sdf_reverted_in_day = sdf_reverted[sdf_reverted.seconds_to_revert.astype('str').astype('int')<86400]
sdf_bots = sdf_reverted_in_day[sdf_reverted_in_day.user_is_bot]
# sdf_bots = revisions from week 1-3 which were reverted within 24 hours

In [7]:
# to do next: find which in df_bots match rev_id of where user_is_bot is true?
sdf_another = df[df.user_is_bot]
sdf_another = sdf_another[sdf_another.is_revert]
#sdf_another = all of the bots that reverted a edit/revision

In [8]:
# sdf_bots holds reverted stuff within 3 weeks by bots
sdf1 = sdf_bots.loc[sdf_bots.is_reverted, ["rev_id", "user_text", "revert_id"]]
sdf1.revert_id = sdf1.revert_id.astype('int')

# sdf_another holds possible reverting changes
sdf2 = sdf_another[["rev_id", 'user_text']]

In [9]:
# subset bot-bot reverts using merge

bot_bot_reverts = pd.merge(sdf1, sdf2, 
             how='inner', 
             left_on='revert_id', 
             right_on='rev_id', 
             suffixes=('', '_reverter')
)

In [10]:
bot_pairs = bot_bot_reverts.groupby(["user_text","user_text_reverter"], as_index = False).count().sort_values("rev_id", ascending=False).reset_index()

In [11]:
bot_pairs["is_self_revert"] = False

for row in bot_pairs.itertuples():
    if row.user_text_reverter == row.user_text:
        bot_pairs.loc[row.Index, "is_self_revert"] = True
        

bot_pairs

index                   user_text      user_text_reverter  rev_id  \
0      24          HBC AIV helperbot5      HBC AIV helperbot5    1185   
1      11                  Cyberbot I              Cyberbot I    1004   
2      13                     Cydebot                 Cydebot     820   
3       5                     AvicBot                 AvicBot     532   
4      19                    EmausBot                EmausBot     360   
5      27                 ListeriaBot             ListeriaBot     298   
6      16                      DatBot      HBC AIV helperbot5     295   
7      17                DeltaQuadBot            DeltaQuadBot     136   
8       7                 ClueBot III             ClueBot III     107   
9      36                 Reports bot             Reports bot      92   
10     34                    MusikBot                MusikBot      78   
11     35                    RMCD bot                RMCD bot      63   
12     28       Lowercase sigmabot II              Cyberbot I      61   
13     26                     Legobot                 Legobot      45   
14      6                      COIBot                  COIBot      31   
15     29       Lowercase sigmabot II              Hazard-Bot      26   
16     41                WOSlinkerBot            WOSlinkerBot      21   
17      1                      Acebot                  Acebot      20   
18      3                   AnomieBOT               AnomieBOT      19   
19     14                     DPL bot                 DPL bot      17   
20     15                DYKUpdateBot            DYKUpdateBot      14   
21      2              Amalthea (bot)          Amalthea (bot)      12   
22     31      Lowercase sigmabot III  Lowercase sigmabot III      11   
23     22             Filedelinkerbot         Filedelinkerbot       8   
24     20              EnterpriseyBot          EnterpriseyBot       7   
25     12                 Cyberbot II             Cyberbot II       6   
26     30       Lowercase sigmabot II   Lowercase sigmabot II       4   
27      0                   AAlertBot               AAlertBot       4   
28     32                     Mathbot                 Mathbot       3   
29     33  MediaWiki message delivery             ClueBot III       2   
30      9             CommonsDelinker         CommonsDelinker       2   
31      4                   AnomieBOT                RMCD bot       2   
32     37                      RonBot                  RonBot       2   
33     39                     SineBot      HBC AIV helperbot5       2   
34     40                     SineBot              Hazard-Bot       2   
35     25              JCW-CleanerBot          JCW-CleanerBot       1   
36     23                  GreenC bot              GreenC bot       1   
37     18                    EmausBot                 Bot1058       1   
38     10          Community Tech bot      Community Tech bot       1   
39      8                  ClueBot NG                EmausBot       1   
40     38                     RussBot              Hazard-Bot       1   
41     21                   Femto Bot               Femto Bot       1   

    revert_id  rev_id_reverter  is_self_revert  
0        1185             1185            True  
1        1004             1004            True  
2         820              820            True  
3         532              532            True  
4         360              360            True  
5         298              298            True  
6         295              295           False  
7         136              136            True  
8         107              107            True  
9          92               92            True  
10         78               78            True  
11         63               63            True  
12         61               61           False  
13         45               45            True  
14         31               31            True  
15         26               26           False  
16         21               21 

In [12]:
bot_bot_reverts[bot_bot_reverts.user_text == "AvicBot"] #ALL SELF-REVERTS

rev_id user_text  revert_id  rev_id_reverter user_text_reverter
46    876243659   AvicBot  876271234        876271234            AvicBot
47    876244058   AvicBot  876271234        876271234            AvicBot
48    876244450   AvicBot  876271234        876271234            AvicBot
49    876244893   AvicBot  876271234        876271234            AvicBot
50    876245314   AvicBot  876271234        876271234            AvicBot
...         ...       ...        ...              ...                ...
5244  879508548   AvicBot  879511314        879511314            AvicBot
5245  879510573   AvicBot  879511314        879511314            AvicBot
5269  879511290   AvicBot  879518555        879518555            AvicBot
5270  879514610   AvicBot  879519188        879519188            AvicBot
5282  879530797   AvicBot  879533764        879533764            AvicBot

[532 rows x 5 columns]

Now exploirng TOP 10 reverted and "reverting" edits 
==


In [13]:
bot_bot_reverts[bot_bot_reverts.user_text == "AvicBot"].head(20)

rev_id user_text  revert_id  rev_id_reverter user_text_reverter
46   876243659   AvicBot  876271234        876271234            AvicBot
47   876244058   AvicBot  876271234        876271234            AvicBot
48   876244450   AvicBot  876271234        876271234            AvicBot
49   876244893   AvicBot  876271234        876271234            AvicBot
50   876245314   AvicBot  876271234        876271234            AvicBot
51   876254649   AvicBot  876271234        876271234            AvicBot
135  876267825   AvicBot  876268182        876268182            AvicBot
151  876270552   AvicBot  876277467        876277467            AvicBot
152  876270563   AvicBot  876275873        876275873            AvicBot
235  876286171   AvicBot  876289513        876289513            AvicBot
236  876287145   AvicBot  876289513        876289513            AvicBot
237  876287797   AvicBot  876289513        876289513            AvicBot
238  876289121   AvicBot  876289513        876289513            AvicBot
258  876289513   AvicBot  876295082        876295082            AvicBot
259  876292591   AvicBot  876295082        876295082            AvicBot
260  876293533   AvicBot  876295082        876295082            AvicBot
261  876294173   AvicBot  876295082        876295082            AvicBot
294  876300057   AvicBot  876300758        876300758            AvicBot
309  876308853   AvicBot  876314649        876314649            AvicBot
310  876309523   AvicBot  876314649        876314649            AvicBot

THESE ARE THE FIRST 20 REVISIONS WITHIN THESE THREE WEEKS!


* rev id: 876243659 - (according to the edit summary) AvicBot was adding and deleting candidates for speedy deletion as abandoned drafrts or AfC aubmissions; this revert is just showed when the page was back to 0 entries 

* rev id: 876244058 - This edit summary was the same as the previous one, yet the revisions don't seem to be related and don't even seem to be reverts... 

*  rev id: 876244450 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg", although it didn't orginially start empty)

*  rev id: 876244893 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg", although it didn't orginially start empty), 9 to 13 to 0 entries

*  rev id: 876245314 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg", although it didn't orginially start empty) 13 to 19 to 0

*  rev id: 876254649 - Again this edit summary wwas the same as the previous one, yet it doesn't seem to be an exact revert (what made this a 'revert' was the fact "Draft:Don Strasburg", although it didn't orginially start empty) 19 to 1 to 0

*  rev id: 876267825 - This edit summary was "Listifying from Category:Candidates for speedy deletion". A user was added and that same user was deleted, exact self revert

*  rev id: 876270552 -  This edit summary was "Listifying from Category:Wikipedia usernames with possible policy issues". A user was added and two user talk pages switched positions, then  that same user was deleted and those same user talk pages switched backed to their orignal positions, exact self revert

*  rev id: 876270563 - This edit summary was the same as the last one. The revision was the same as the previosu rev id as well: A user was added and two user talk pages switched positions, then  that same user was deleted and those same user talk pages switched backed to their orignal positions, exact self revert. The only difference is when the original and last edit was (different time yet same content and entries as original/last entry for previous rev id)

*  rev id: 876286171 - Again, AvicBot was "Listifying from Category:Candidates for speedy deletion". Allthoguh the edits seem to be somewhat related, it doesn't seem like a revert. 4 to 6 to 4 entries (same 4 entries that the original edit started with) 
*  rev id: 876287145 - This has the same edit summary as above, and while a user was added to this category at the time (6 to 7 entries), the revision that reverted it from 3 to 4 entries. the revisions in between removed three users (1 that was added in the orignal edit)

*  rev id: 876287797 - This has the same edit summary as above, and while a user was added to this category (one page switch positions) at the time (7 to 8 entries), the revision that reverted it from 3 to 4 entries. this revisions removed 4 users (a page being removed before the original edit and being added again )

*  rev id: 876289121 - This has the same edit summary as above, and 5 entries were removed during the original edit (lef twith 3 entries), the revision that reverted it from 3 to 4 entries. the revisions in between removed four users (1 that was added and swicthed in the orignal edit)

*  rev id: 876289513 - This has the same idt summary as above, where a user was added to hte original edit (3 to 4 entries). the revision that reverted it removed a user that was added by revision in between, back to the original 3 entries

*  rev id: 876292591 - This has the same edit summary as above, these edits don't seem to be reverts, though. the original edit start with 4 entries and then a catergory was added, but the revision that reverted this removed a user that was added by revision in between (not the category that was added before), but not back to its original edit

*  rev id: 876293533 - This has the same edit summary as above. the original edit start with 5 entries and then a page was added and a couple of pages switched positions; the revision that reverted this removed the page that was added earlier, but not back to its original edit

*  rev id: 876294173 - This has the same edit summary as above, these edits don't seem to be reverts, though. the original edit start with 6 entries and then two pages were removed and a page switched positions; the revision that reverted this removed the page that was added by another revision, but not back to its original edit

*  rev id: 876300057 - (according to the edit summary) AvicBot was adding and deleting candidates for speedy deletion as abandoned drafrts or AfC aubmissions; this revert is just showed when the page was back to 0 entries, 16 revisions added and removed

*  rev id: 876308853 - Again, AvicBot was "Listifying from Category:Candidates for speedy deletion". a page was added to the original edit and the revision that reverted it removed 4 entries that were added by revisions in between it (not the user that was added to the original edit earlier)

*  rev id: 876243659 - This edit summary was the same as the one above. another page was added to the user the edit ABOVE, and the reviison that reverted it removed 4 entries (including the user page that was added earlier) 

NOTES:

* Although all the revision id's here are different, as expected, some of the revert ids are the same, even thought the revision id is different - this is because all of those revisions have the same revert id, they are apart of revisions that happened in between reverts that weren't exact. 

* looked at these pages and are no longer in use as of July 2020

Questions: 
==

* Why are there adminstrator newsletters on this talk page? (only of months April, May, and June, and sent by MediaWiki message delivery 
* Wonder how people are notified of these discussions, and if users can request the results of trials or wants to give feedback to a bot/point out a mistake on one of these discussions that could change the result of the overall discussion?
* Why does a bot decide to swtich positions of user talk pages? what is the order based on? 